statsmodels/base/data.py

```diff
@@ -76,6 +80,26 @@ def __init__(self, endog, exog=None, missing='none', hasconst=None,
         self._check_integrity()
         self._cache = resettable_cache()
 
+    def __getstate__(self):
+        from copy import copy
+        d = copy(self.__dict__)
+        if "design_info" in d:
+            del d["design_info"]
+            d["restore_design_info"] = True
+        return d
+
+    def __setstate__(self, d):
+        if "restore_design_info" in d:
+            # NOTE: there may be a more performant way to do this
+            from patsy import dmatrices
+            depth = 1  # hmm, have to have same eval env in calling ns
+            data = d['orig_endog'].join(d['orig_exog'])
+            _, design = dmatrices(d['formula'], data, depth,
```

Comment:
use `eval_env=depth`  keyword use of keywords, it's more explicit and less fragile